src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -23,13 +23,27 @@ public static class Program
         public static int Main(string[] args)
         {
             const int expectedArguments = 17;
+            const int expectedArgumentsWithValidateOnly = 18; // Extra --validate-only argument for CI checks.
```

Comment:
I would take a slightly different approach

1. Bump up `expectedArguments` to 18, so we always expect a true or false value for the new bool argument for validation.
2. Default the MSBuild property for this bool to `false` when not set. This will eventually be appended or prepended to the arg list here: https://github.com/dotnet/roslyn-analyzers/blob/9e6acea0f2a9589ab799e3e8664992d6867a21bd/eng/GenerateAnalyzerNuspec.targets#L91
3. Revert all arg parsing/validation changes below and just parse the bool at explicit index into `validateOnly`.